In [1]:
import numpy as np
import pandas as pd

from tqdm.notebook import tqdm
tqdm.pandas()

In [2]:
cares_data_path = "C:/Users/aksha/OneDrive/Documents/mitb_sem2.5/Health_Analytics/Health_Leo/CARES_data_imputedv1.csv"
data = pd.read_csv(cares_data_path)
data.head()

GENDER  RCRI_score  Anemia category  Preoptransfusionwithin30days  Intraop  \
0  FEMALE           0             mild                             0        0   
1    MALE           0  moderate/severe                             0        1   
2    MALE           0             mild                             0        0   
3    MALE           0             none                             0        0   
4  FEMALE           0             none                             0        0   

   Postopwithin30days  Transfusionintraandpostop AnaestypeCategory  \
0                   0                          0                GA   
1                   0                          1                GA   
2                   0                          0                GA   
3                   0                          0                GA   
4                   0                          0                GA   

  PriorityCategory TransfusionIntraandpostopCategory  ... RaceCategory  \
0         Elective                           0 units  ...      Chinese   
1         Elective                            1 unit  ...      Chinese   
2         Elective                           0 units  ...      Chinese   
3        Emergency                           0 units  ...       Indian   
4         Elective                           0 units  ...      Chinese   

  CVARCRICategory  IHDRCRICategory CHFRCRICategory DMinsulinRCRICategory  \
0              no               no              no                    no   
1              no               no              no                    no   
2              no               no              no                    no   
3              no               no              no                    no   
4              no               no              no                    no   

  CreatinineRCRICategory GradeofKidneyCategory  RDW15.7 ASAcategorybinned  \
0                     no                    G1  <= 15.7                 I   
1                     no                    G1  <= 15.7                 I   
2                     no                    G1    >15.7                II   
3                     no                    G1  <= 15.7                 I   
4                     no                    G2  <= 15.7                II   

  ICUAdmgt24h  
0          no  
1          no  
2          no  
3          no  
4          no  

[5 rows x 24 columns]

In [3]:
data.isna().sum()

GENDER                               0
RCRI_score                           0
Anemia category                      0
Preoptransfusionwithin30days         0
Intraop                              0
Postopwithin30days                   0
Transfusionintraandpostop            0
AnaestypeCategory                    0
PriorityCategory                     0
TransfusionIntraandpostopCategory    0
AGEcategory                          0
Mortality                            0
thirtydaymortality                   0
SurgRiskCategory                     0
RaceCategory                         0
CVARCRICategory                      0
IHDRCRICategory                      0
CHFRCRICategory                      0
DMinsulinRCRICategory                0
CreatinineRCRICategory               0
GradeofKidneyCategory                0
RDW15.7                              0
ASAcategorybinned                    0
ICUAdmgt24h                          0
dtype: int64

In [4]:
data.shape

(69667, 24)

In [5]:
data.dtypes

GENDER                               object
RCRI_score                            int64
Anemia category                      object
Preoptransfusionwithin30days          int64
Intraop                               int64
Postopwithin30days                    int64
Transfusionintraandpostop             int64
AnaestypeCategory                    object
PriorityCategory                     object
TransfusionIntraandpostopCategory    object
AGEcategory                          object
Mortality                            object
thirtydaymortality                     bool
SurgRiskCategory                     object
RaceCategory                         object
CVARCRICategory                      object
IHDRCRICategory                      object
CHFRCRICategory                      object
DMinsulinRCRICategory                object
CreatinineRCRICategory               object
GradeofKidneyCategory                object
RDW15.7                              object
ASAcategorybinned               

<b>Gender

In [6]:
# Frequency Count: Counting the number of occurrences for each gender
gender_counts = data['GENDER'].value_counts()
print("Frequency Count for each Gender:")
print(gender_counts)

# Calculating Mortality Rate: Proportion of deaths for each gender
mortality_rate = data.groupby('GENDER')['ICUAdmgt24h'].apply(lambda x: (x == 'yes').mean())
print("\nMortality Rate for each Gender:")
print(mortality_rate)

Frequency Count for each Gender:
GENDER
FEMALE    35099
MALE      34568
Name: count, dtype: int64

Mortality Rate for each Gender:
GENDER
FEMALE    0.012564
MALE      0.021060
Name: ICUAdmgt24h, dtype: float64


<b>Chi-Square Test of Independence<br>
Null Hypothesis (H0): There is no association between gender and mortality; they are independent.<br>
Alternative Hypothesis (H1): There is an association between gender and mortality; they are not independent.

In [7]:
from scipy import stats

In [8]:
# Create a contingency table
contingency_table = pd.crosstab(data['GENDER'], data['ICUAdmgt24h'])
print("Contingency Table:")
print(contingency_table)

Contingency Table:
ICUAdmgt24h     no  yes
GENDER                 
FEMALE       34658  441
MALE         33840  728


In [9]:
# Perform Chi-Square Test of Independence
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)

In [10]:
print("\nChi-Square Test Results:")
print(f"Chi2 Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")
print("Expected Frequencies:") #Expected frequencies (if there were no relationship between gender and mortality) 
print(expected)


Chi-Square Test Results:
Chi2 Statistic: 75.67269323993298
P-value: 3.348058845829266e-18
Degrees of Freedom: 1
Expected Frequencies:
[[34510.04495672   588.95504328]
 [33987.95504328   580.04495672]]


<b>The p-value is significantly smaller than the common significance level (e.g., 0.05), indicating that the observed association between gender and ICUAdmgt24h is statistically significant.

<b>RCRI Score

In [18]:
# Converting Mortality column to a categorical type with explicit categories
data['ICUAdmgt24h'] = pd.Categorical(data['ICUAdmgt24h'], categories=['no', 'yes'])

# Summary Statistics for RCRI_score
mean_rcri = data['RCRI_score'].mean()
median_rcri = data['RCRI_score'].median()
std_rcri = data['RCRI_score'].std()
range_rcri = data['RCRI_score'].max() - data['RCRI_score'].min()

print("Summary Statistics for RCRI Score:")
print(f"Mean: {mean_rcri}")
print(f"Median: {median_rcri}")
print(f"Standard Deviation: {std_rcri}")
print(f"Range: {range_rcri}")

# Mortality Rate Calculation for each RCRI score
mortality_rate = data.groupby('RCRI_score')['ICUAdmgt24h'].apply(lambda x: (x == 'yes').mean())
print("\nICUAdmgt24h Rate for each RCRI Score:")
print(mortality_rate)

Summary Statistics for RCRI Score:
Mean: 0.2744197396184707
Median: 0.0
Standard Deviation: 0.6091160443592442
Range: 6

ICUAdmgt24h Rate for each RCRI Score:
RCRI_score
0    0.010378
1    0.031407
2    0.064216
3    0.100000
4    0.087379
5    0.121212
6    0.000000
Name: ICUAdmgt24h, dtype: float64


<b>Logistic regression will allow us to assess the odds of ICUAdmgt24h as a function of the RCRI score and understand how the risk changes with different scores.<br>
<i>Null Hypothesis (H0): The RCRI score does not predict the odds of ICUAdmgt24h.<br>
Alternative Hypothesis (H1): The RCRI score predicts the odds of ICUAdmgt24h.

In [19]:
# Convert 'ICUAdmgt24h' to a binary numeric variable (0 for 'No death', 1 for 'Yes')
data['ICUAdmgt24h_numeric'] = (data['ICUAdmgt24h'] == 'yes').astype(int)

In [20]:
import statsmodels.api as sm

# Logistic Regression
X = sm.add_constant(data['RCRI_score'])  # Adding a constant (intercept)
y = data['ICUAdmgt24h_numeric']

model = sm.Logit(y, X)
result = model.fit()

print(result.summary())

Optimization terminated successfully.
         Current function value: 0.081140
         Iterations 8
                            Logit Regression Results                           
Dep. Variable:     ICUAdmgt24h_numeric   No. Observations:                69667
Model:                           Logit   Df Residuals:                    69665
Method:                            MLE   Df Model:                            1
Date:                 Wed, 14 Aug 2024   Pseudo R-squ.:                 0.04796
Time:                         15:16:42   Log-Likelihood:                -5652.8
converged:                        True   LL-Null:                       -5937.5
Covariance Type:             nonrobust   LLR p-value:                7.322e-126
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -4.4321      0.036   -122.466      0.000      -4.503      -4.361
RCRI_score     0.7541

<b>RCRI Score: Each unit increase in the RCRI score increases the log-odds of mortality by approximately 0.7058.<br>
Odds Ratio: exp(0.7541) ≈ 2.12<br>
This means that for each additional point in the RCRI score, the odds of mortality are approximately 2 times higher.<br>

<b>Anemia Category

In [21]:
# Frequency Count for each Anemia Category
frequency_count = data['Anemia category'].value_counts()
print("Frequency Count for each Anemia Category:")
print(frequency_count)

# Mortality Rate Calculation for each Anemia Category
mortality_rate = data.groupby('Anemia category')['ICUAdmgt24h_numeric'].mean()
print("\nICUAdmgt24h Rate for each Anemia Category:")
print(mortality_rate)

Frequency Count for each Anemia Category:
Anemia category
none               48116
mild               11590
moderate/severe     9961
Name: count, dtype: int64

ICUAdmgt24h Rate for each Anemia Category:
Anemia category
mild               0.021829
moderate/severe    0.043670
none               0.009997
Name: ICUAdmgt24h_numeric, dtype: float64


In [22]:
# Convert Mortality column to a binary numeric variable (0 for 'No death', 1 for 'Yes')
# data['Mortality_numeric'] = (data['Mortality'] == 'Yes').astype(int)

# Create a contingency table
contingency_table = pd.crosstab(data['Anemia category'], data['ICUAdmgt24h_numeric'])

# Perform Chi-Square Test
chi2_stat, p_value, dof, expected = stats.chi2_contingency(contingency_table)

# Print results
print("Chi-Square Statistic:", chi2_stat)
print("P-value:", p_value)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:")
print(expected)

# Interpret results
alpha = 0.05
if p_value < alpha:
    print("There is a significant association between Anemia category and ICUAdmgt24h.")
else:
    print("There is no significant association between Anemia category and ICUAdmgt24h.")

Chi-Square Statistic: 588.677248368308
P-value: 1.4803339720113876e-128
Degrees of Freedom: 2
Expected Frequencies:
[[11395.52183961   194.47816039]
 [ 9793.85617294   167.14382706]
 [47308.62198745   807.37801255]]
There is a significant association between Anemia category and ICUAdmgt24h.


<b>Preop Transfusion within 30 Days

In [23]:
# Convert Mortality column to a binary numeric variable (0 for 'No death', 1 for 'Yes')
# data['Mortality_numeric'] = (data['Mortality'] == 'Yes').astype(int)

# Frequency Count for Preoperative Transfusion
frequency_count = data['Preoptransfusionwithin30days'].value_counts()
print("Frequency Count for Preoperative Transfusion:")
print(frequency_count)

Frequency Count for Preoperative Transfusion:
Preoptransfusionwithin30days
0     67882
1       958
2       436
3       176
4        73
5        44
7        23
6        23
8        19
10        7
11        7
9         6
13        5
14        4
17        1
12        1
20        1
21        1
Name: count, dtype: int64


In [24]:
from scipy.stats import fisher_exact

# Combine transfusion categories into two groups
data['Transfusion_Group'] = data['Preoptransfusionwithin30days'].apply(lambda x: 'No transfusion' if x == 0 else 'Any transfusion')

# Create a 2x2 contingency table
contingency_table = pd.crosstab(data['Transfusion_Group'], data['ICUAdmgt24h_numeric'])

# Perform Fisher's Exact Test
oddsratio, p_value = fisher_exact(contingency_table)

print("Fisher's Exact Test Results:")
print(f"Contingency Table:\n{contingency_table}")
print(f"Odds Ratio: {oddsratio}")
print(f"P-value: {p_value}")

Fisher's Exact Test Results:
Contingency Table:
ICUAdmgt24h_numeric      0    1
Transfusion_Group              
Any transfusion       1426  359
No transfusion       67072  810
Odds Ratio: 0.047969902879074613
P-value: 3.035323234247888e-284


Explanation:<br>

Any transfusion: Patients who received at least one unit of transfusion.<br>
1187: Number of patients in this group who did not die.<br>
598: Number of patients in this group who died.<br><br>
No transfusion: Patients who received no transfusion.<br>
63140: Number of patients in this group who did not die.<br>
4742: Number of patients in this group who died.<br><br>
<b>The p-value is extremely small which indicates that there is a very strong statistical evidence against the null hypothesis.<br>
The odds of mortality in the transfusion group are about 0.05 times (or 5%) of the odds in the no transfusion group, suggesting a protective effect of transfusions against ICUAdmgt24h.

<b>Transfusionintraandpostop

In [ ]:
# Convert Mortality column to a binary numeric variable (0 for 'No death', 1 for 'Yes')
# data['Mortality_numeric'] = (data['Mortality'] == 'Yes').astype(int)

# Frequency Count for Preoperative Transfusion
frequency_count = data['Transfusionintraandpostop'].value_counts()
print("Frequency Count for Transfusion intra and post operation:")
print(frequency_count)

In [25]:
from scipy.stats import fisher_exact

# Combine transfusion categories into two groups
data['Transfusion_Group'] = data['Transfusionintraandpostop'].apply(lambda x: 'No transfusion' if x == 0 else 'Any transfusion')

# Create a 2x2 contingency table
contingency_table = pd.crosstab(data['Transfusion_Group'], data['ICUAdmgt24h_numeric'])

# Perform Fisher's Exact Test
oddsratio, p_value = fisher_exact(contingency_table)

print("Fisher's Exact Test Results:")
print(f"Contingency Table:\n{contingency_table}")
print(f"Odds Ratio: {oddsratio}")
print(f"P-value: {p_value}")

Fisher's Exact Test Results:
Contingency Table:
ICUAdmgt24h_numeric      0    1
Transfusion_Group              
Any transfusion       4225  489
No transfusion       64273  680
Odds Ratio: 0.09141094431132639
P-value: 1.608392066248614e-258


Explanation:<br>

Any transfusion: Patients who received at least one unit of transfusion.<br>
3520: Number of patients who did not die.<br>
1194: Number of patients who died.<br>
No transfusion: Patients who received no transfusion.<br>
60807: Number of patients who did not die.<br>
4146: Number of patients who died.<br><br>
<b>The p-value is extremely small which indicates that there is a very strong statistical evidence against the null hypothesis.<br>
The odds of mortality for patients with any transfusion are about 9% of the odds for patients with no transfusion, suggesting a protective effect of transfusion against ICUAdmgt24h.

<b>AnaestypeCategory

In [26]:
# Frequency count of each anesthesia type
frequency_count = data['AnaestypeCategory'].value_counts()

# Mortality rate for each anesthesia type
mortality_rate = data.groupby('AnaestypeCategory')['ICUAdmgt24h_numeric'].mean()

# Display the results
print("Frequency Count for each Anesthesia Type:")
print(frequency_count)
print("\nMortality Rate for each Anesthesia Type:")
print(mortality_rate)

Frequency Count for each Anesthesia Type:
AnaestypeCategory
GA    58364
RA    11303
Name: count, dtype: int64

Mortality Rate for each Anesthesia Type:
AnaestypeCategory
GA    0.019087
RA    0.004866
Name: ICUAdmgt24h_numeric, dtype: float64


<b>Frequency: General anesthesia is much more common than regional anesthesia in this dataset.<br>
Mortality Rate: The mortality rate is higher for patients who underwent surgery with regional anesthesia compared to those with general anesthesia (~2% vs. ~0.4%).

In [27]:
# Create a contingency table
contingency_table = pd.crosstab(data['AnaestypeCategory'], data['ICUAdmgt24h_numeric'])

# Perform Chi-Square test
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)

# Display the results
print("\nChi-Square Test Results:")
print(f"Chi2 Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")


Chi-Square Test Results:
Chi2 Statistic: 115.21564853140003
P-value: 7.058838055594819e-27
Degrees of Freedom: 1


<b>There is a statistically significant association between the type of anesthesia and ICUAdmgt24h rates.

<b>Race Category

In [28]:
frequency_count = data['RaceCategory'].value_counts()
print("Frequency Count for each Priority Category:")
print(frequency_count)

Frequency Count for each Priority Category:
RaceCategory
Chinese    48605
Malay       7431
Others      7019
Indian      6612
Name: count, dtype: int64


In [29]:
# Contingency Table
contingency_table = pd.crosstab(data['RaceCategory'], data['ICUAdmgt24h_numeric'])
print("\nContingency Table:")
print(contingency_table)

# Chi-Square Test
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("\nChi-Square Test Results:")
print(f"Chi2 Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")


Contingency Table:
ICUAdmgt24h_numeric      0    1
RaceCategory                   
Chinese              47759  846
Indian                6507  105
Malay                 7320  111
Others                6912  107

Chi-Square Test Results:
Chi2 Statistic: 4.0100259487504575
P-value: 0.2603835418212841
Degrees of Freedom: 3


<b> There is a no significant association between the Race and patient mortality. The p-value is more than 0.05 indicating lack of sufficient evidence to reject Null hypothesis.

<b>Age Category

In [30]:
# Frequency Count
age_category_count = data['AGEcategory'].value_counts()
print("Frequency Count for each Age Category:")
print(age_category_count)

# Mortality Rate
age_category_mortality_rate = data.groupby('AGEcategory')['ICUAdmgt24h_numeric'].mean()
print("\nMortality Rate for each Age Category:")
print(age_category_mortality_rate)

Frequency Count for each Age Category:
AGEcategory
50-64    22110
30-49    19419
65-74    13139
18-29     6983
75-84     6724
>=85      1292
Name: count, dtype: int64

Mortality Rate for each Age Category:
AGEcategory
18-29    0.005728
30-49    0.008703
50-64    0.015920
65-74    0.024355
75-84    0.034949
>=85     0.041022
Name: ICUAdmgt24h_numeric, dtype: float64


In [31]:
# Contingency Table
contingency_table_age = pd.crosstab(data['AGEcategory'], data['ICUAdmgt24h_numeric'])
print("\nContingency Table:")
print(contingency_table_age)

# Chi-Square Test
chi2_age, p_age, dof_age, expected_age = stats.chi2_contingency(contingency_table_age)
print("\nChi-Square Test Results:")
print(f"Chi2 Statistic: {chi2_age}")
print(f"P-value: {p_age}")
print(f"Degrees of Freedom: {dof_age}")


Contingency Table:
ICUAdmgt24h_numeric      0    1
AGEcategory                    
18-29                 6943   40
30-49                19250  169
50-64                21758  352
65-74                12819  320
75-84                 6489  235
>=85                  1239   53

Chi-Square Test Results:
Chi2 Statistic: 355.7421570463822
P-value: 1.0156666132806244e-74
Degrees of Freedom: 5


<b> There is a statistically significant association between age and patient mortality.

<b> ASA category

In [32]:
# Frequency Count
asa_category_count = data['ASAcategorybinned'].value_counts()
print("Frequency Count for each ASA Category:")
print(asa_category_count)

# Mortality Rate
asa_category_mortality_rate = data.groupby('ASAcategorybinned')['ICUAdmgt24h_numeric'].mean()
print("\nICUAdmgt24h Rate for each ASA Category:")
print(asa_category_mortality_rate)

Frequency Count for each ASA Category:
ASAcategorybinned
II       39286
I        15870
III      13420
IV-VI     1091
Name: count, dtype: int64

ICUAdmgt24h Rate for each ASA Category:
ASAcategorybinned
I        0.002773
II       0.008960
III      0.038450
IV-VI    0.235564
Name: ICUAdmgt24h_numeric, dtype: float64


In [33]:
# Contingency Table
contingency_table_asa = pd.crosstab(data['ASAcategorybinned'], data['ICUAdmgt24h_numeric'])
print("\nContingency Table:")
print(contingency_table_asa)

# Chi-Square Test
chi2_asa, p_asa, dof_asa, expected_asa = stats.chi2_contingency(contingency_table_asa)
print("\nChi-Square Test Results:")
print(f"Chi2 Statistic: {chi2_asa}")
print(f"P-value: {p_asa}")
print(f"Degrees of Freedom: {dof_asa}")


Contingency Table:
ICUAdmgt24h_numeric      0    1
ASAcategorybinned              
I                    15826   44
II                   38934  352
III                  12904  516
IV-VI                  834  257

Chi-Square Test Results:
Chi2 Statistic: 3881.644377419353
P-value: 0.0
Degrees of Freedom: 3


<b> There is a statistically significant association between ASA category and patient mortality.

## Multivariate Logistic to factor in all

Logistic Regression Details
Single Dependent Variable: In logistic regression, even though the dependent variable (e.g., mortality) is binary (Yes/No), the model considers the effects of multiple independent variables simultaneously. This is why logistic regression is considered a multivariate technique.

Independent Variables: These can include a mix of categorical and continuous predictors. The goal is to understand how each predictor, while accounting for the presence of others, affects the probability of the outcome.

In [34]:
from sklearn.preprocessing import LabelEncoder

# Convert categorical variables to numeric using Label Encoding
label_encoders = {}
for col in ['GENDER', 'Anemia category', 'AnaestypeCategory', 'PriorityCategory',
            'TransfusionIntraandpostopCategory', 'AGEcategory', 'SurgRiskCategory',
            'RaceCategory', 'CVARCRICategory', 'IHDRCRICategory', 'CHFRCRICategory',
            'DMinsulinRCRICategory', 'CreatinineRCRICategory', 'GradeofKidneyCategory',
            'RDW15.7', 'ASAcategorybinned', 'Mortality']:
    le = LabelEncoder()
    data[col] = le.fit_transform(data[col].astype(str))  # Ensure all data is string for encoding
    label_encoders[col] = le

# Convert target variable 'ICUAdmgt24h' to binary
data['ICUAdmgt24h_numeric'] = (data['ICUAdmgt24h'] == 'yes').astype(int)

# Define the independent and dependent variables
X = data[['RCRI_score', 'Preoptransfusionwithin30days', 'Intraop', 'Postopwithin30days', 
          'Transfusionintraandpostop', 'AnaestypeCategory', 'PriorityCategory', 
          'TransfusionIntraandpostopCategory', 'AGEcategory', 'SurgRiskCategory', 
          'RaceCategory', 'CVARCRICategory', 'IHDRCRICategory', 'CHFRCRICategory', 
          'DMinsulinRCRICategory', 'CreatinineRCRICategory', 'GradeofKidneyCategory', 
          'RDW15.7', 'ASAcategorybinned', 'Mortality']]

y = data['ICUAdmgt24h_numeric']

# Add a constant to the model (intercept)
X = sm.add_constant(X)

# Ensure there are no missing values
X = X.dropna()
y = y.loc[X.index]  # Align y with X

# Fit the logistic regression model
model = sm.Logit(y, X)
result = model.fit()

# Print the summary of the logistic regression
print(result.summary())

Optimization terminated successfully.
         Current function value: 0.061554
         Iterations 10
                            Logit Regression Results                           
Dep. Variable:     ICUAdmgt24h_numeric   No. Observations:                69667
Model:                           Logit   Df Residuals:                    69647
Method:                            MLE   Df Model:                           19
Date:                 Wed, 14 Aug 2024   Pseudo R-squ.:                  0.2778
Time:                         16:51:31   Log-Likelihood:                -4288.2
converged:                        True   LL-Null:                       -5937.5
Covariance Type:             nonrobust   LLR p-value:                     0.000
                                        coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------
const                                -6.1680      0.1

Note:<br><b>Chi-Square Tests: These tests assess the association between two variables independently. They do not account for other variables that might also be influencing the outcome.
<b><br>
Logistic Regression: When multiple variables are included in a logistic regression model, the effect of each variable is assessed while holding other variables constant. If some variables are highly correlated with each other (multicollinearity), it can affect the coefficients and significance levels of individual predictors.

In the logistic regression results, variables with p-values less than the conventional significance level of 0.05 are considered statistically significant. In this case, variables such as RCRI_score, Preoptransfusionwithin30days, AnaestypeCategory, PriorityCategory, TransfusionIntraandpostopCategory, AGEcategory, SurgRiskCategory, RaceCategory, CVARCRICategory, IHDRCRICategory, CHFRCRICategory, DMinsulinRCRICategory, CreatinineRCRICategory, GradeofKidneyCategory, ASAcategorybinned, and Mortality are significant, with p-values well below 0.05. This suggests that these variables have a meaningful association with the likelihood of mortality, meaning their effects are not likely due to random chance.

On the other hand, variables such as Intraop, Postopwithin30days, Transfusionintraandpostop, RDW15.7 either have missing values or p-values greater than 0.05, suggesting that they do not have a statistically significant impact on mortality in this model. These non-significant results might differ from individual chi-square tests because the logistic regression model accounts for the effects of all predictors simultaneously, allowing for the assessment of each variable's effect while controlling for others. Chi-square tests, on the other hand, evaluate the association between each variable and mortality independently, which can sometimes lead to different conclusions due to the lack of control for confounding variables.

<b>ANOVA

In [35]:
import pandas as pd
from scipy.stats import kruskal

# Define the DataFrame
# data = pd.read_csv('your_data_file.csv')

# Define the categorical variable and continuous variables
categorical_var = 'ICUAdmgt24h'
continuous_vars = [
    'RCRI_score',
    'Preoptransfusionwithin30days',
    'Intraop',
    'Postopwithin30days',
    'Transfusionintraandpostop'
]

def perform_kruskal_wallis(data, categorical_var, continuous_vars):
    results = {}
    
    # Get the unique groups in the categorical variable
    groups = data[categorical_var].unique()
    
    for var in continuous_vars:
        # Extract data for each group
        group_data = [data[data[categorical_var] == group][var].dropna() for group in groups]
        
        # Perform Kruskal-Wallis test
        kruskal_result = kruskal(*group_data)

        # Store the results
        results[var] = {
            'H-statistic': kruskal_result.statistic,
            'p-value': kruskal_result.pvalue
        }
    
    return results

# Perform the test
kruskal_wallis_results = perform_kruskal_wallis(data, categorical_var, continuous_vars)

# Print the results
for var, result in kruskal_wallis_results.items():
    print(f"Kruskal-Wallis test for {var}:")
    print(f"  H-statistic = {result['H-statistic']:.4f}")
    print(f"  p-value = {result['p-value']:.9f}")
    print("-" * 40)

Kruskal-Wallis test for RCRI_score:
  H-statistic = 732.7198
  p-value = 0.000000000
----------------------------------------
Kruskal-Wallis test for Preoptransfusionwithin30days:
  H-statistic = 3833.2621
  p-value = 0.000000000
----------------------------------------
Kruskal-Wallis test for Intraop:
  H-statistic = 2313.5680
  p-value = 0.000000000
----------------------------------------
Kruskal-Wallis test for Postopwithin30days:
  H-statistic = 419.3417
  p-value = 0.000000000
----------------------------------------
Kruskal-Wallis test for Transfusionintraandpostop:
  H-statistic = 2317.3586
  p-value = 0.000000000
----------------------------------------


<b>Interpretation:</b><br>

There is a significant difference in all the considered continuous variables among yes/no groups in ICUAdmgt24h since p-value <0.05 (null hypothesis is rejected)